### PR TITLE
Temporarily remove the 'popup' feature

### LINKED ISSUE #6290: Highlighting leads to options in code editor. Would Shift + highlight be better
Hey guys,

I just noticed a new feature in the code editor today. Its great to have for copying and pasting code to ask questions, but it can get in the way if I am just doing regular copy and pasting.

Would a better solution to this be to hold down shift + highlight the section as a way to access the menu?

What do you guys think?

### LINKED ISSUE #6587: "Copy as..." overzealous popup
The new "Copy {Link, as Pretty Code, as Plain Code}" popup feature appears even when I am just trying to highlight part of a line to delete at once. As a result, it is almost always on my screen, and my feeling is that it would be the right move to offer a "Never show this again" option or at least some way to switch it off.

### PR DESCRIPTION
**Pull request by bugron**
Tested locally.
Closes #6587, closes #6290 and closes #6320.

### CODE REVIEW

**Comment by alistermada** [comment]
Hmm if this is implemented, I wonder how many people will realize this feature still exists but is triggered a bit differently. Most might assume it's gone I'm guessing. 

Maybe add a little reminder somewhere on how the tooltip is opened?

File: client/commonFramework/create-editor.js [hunk 1]
```diff
@@ -109,8 +109,10 @@ contents of code editor'>Copy as Plain Code</a>"></a>`;
     };
   }
 
-  var showPopover = debounce(function() {
-    if (editor.somethingSelected()) {
+  var showPopover = debounce(function(e) {
+    var isCtrlKey = e ? e.ctrlKey : true,
+    isAltKey = e ? e.altKey : true;
+    if (editor.somethingSelected() && (isCtrlKey || isAltKey)) {
       $('#copy-btn-parent').popover('show');
```
Comment (BerkeleyTrue) [inline client/commonFramework/create-editor.js]: Lets keep each logical expression on its own line.

```js
if (
  editor.somethingSelected() &&
  (isCtrlKey || isAltKey)
) {
```

**Comment by SaintPeter** [comment]
I've been thinking about discoverability.  When text is selected, I'd like to see a small bit of text under the Codemirror which says: 
> Hold CTRL when selecting for copy menu

Or something like that.  It should only display when text is selected.

**Comment by bugron** [comment]
@SaintPeter I'm sure that will be annoying for campers too to see that every time.

File: client/commonFramework/create-editor.js (hunk 1 above)
Comment (bugron) [inline client/commonFramework/create-editor.js]: Will do, thanks.

**Comment by BerkeleyTrue** [comment]
> I'm sure that will be annoying for campers too to see that every time.

Agreed.

**Comment by abhisekp** [comment]
**Proposed Solution:** Show the popover when COPY event is fired.

**Comment by bugron** [comment]
@abhisekp then why do we need this?

**Comment by abhisekp** [comment]
@bugron because normal copy doesn't *pretty copy* or *copy the link*.

And about `Copy Link`, show as an item of the list and also add a icon at the end of the challenge title in the left sidebar (with a tooltip for desktop users OR as labelled icon for touch screen users).

And when implementing the `copy` event listener do not prevent default or stop propagation.

**Comment by AkiraLaine** [comment]
I'm just wondering - on selection, can't we have a small "info" icon that pops up on the top right or something? When the user clicks on it, then we show the different actions. It's less intrusive and clean.

**Comment by bugron** [comment]
So that is why we have this toolpip.
>And about Copy Link, show as an item of the list and also add a icon at the end of the challenge title in the left sidebar (with a tooltip for desktop users OR as labelled icon for touch screen users).
What does this mean? What icon? Any example?

**Comment by AkiraLaine** [comment]
@bugron Well if people think the current tooltip is too intrusive, how about moving the menu thing to the top-right of the editor on selection?

**Comment by nigeljoy** [comment]
May be make that popover bubble appear at the top right corner of code editor when highlighted. 

Pretty sure, nobody here are crazy enoguh to write codes that long in the first few lines.

**Comment by SaintPeter** [comment]
Here is what I imagine:
image

The text would only appear when you have selected or are selecting text without holding the Ctrl key.  Basically whenever the popup would display CURRENTLY, display this text instead, in this location.  If they hold CTRL, don't display the text, instead display the popover.

**Comment by santakmishra** [comment]
@SaintPeter That's an elegant solution.
I was also thinking, how about retaining the existing pop-over thing, but making it optional by putting an option in the Account Settings to set the flag. That way, whoever wants that pretty pop, can turn it on or leave it off if they prefer the normal Cmd+C. Also, the option can be "on" by default and new campers can be informed about it during the first challenge, with an option to turn it off if they want.

**Comment by bugron** [comment]
@SaintPeter just pushed a commit that includes your suggestion. Let me know if I've done anything wrong.

**Comment by SaintPeter** [comment]
I was hoping to get it right below the CodeMirror so it would be more immediately visible.  I only saw it because I was looking around the screen for it.
It also does not disappear when the code is no longer selected.

**Comment by bugron** [comment]
@SaintPeter it is right below the editor for me. Can you comment here a screenshot, please?
I thought that showing/hiding that every time will annoy campers, will fix.

**Comment by SaintPeter** [comment]
image

I think that having it pop up (and go away) will give it the visibility that we need without being obtrusive/blocking content like the current pop-over.

**Comment by bugron** [comment]
@SaintPeter updated.

**Comment by SaintPeter** [comment]
That looks perfect.   I'll defer to @hallaathrad on any additional styling, but I really like the functionality.

**Comment by hallaathrad** [comment]
Test driving on my mac I can't seem to get it to work. Please leat me know if you want to guinea pig away with me. :)

----
Thus far, I can tell you pressing alt brings out a crosshair you use to select items on OSX, and then Ctrl has different behaviours with the mouse and keyboard, including right click and some multitasking stuff.

Earlier tonight, I've mentioned to @SaintPeter how maybe you could use the context menu for this purpose, Thus you'd avoid the custom UI pattern, and crossplatform compatibility

**Comment by QuincyLarson** [comment]
I will be blunt:
- the control-A shortcut used here collides with the OSX hotkeys I use (control+a on mac goes to the beginning of a line). I can do this everywhere on a mac, but suddenly not in FCC's code editor.
- The message popping up every time I highlight something is astonishing.
- Both control and alt in OSX bring up a crosshair cursor.

In my opinion, this feature is not ready for primetime, and we've gotten a ton of complaints about it through email. I humbly suggest we remove this context menu feature completely and go back to the drawing board with a more graceful way to handle this.

**Comment by AkiraLaine** [comment]
@QuincyLarson Maybe open an Issue and have other Campers feedback on how we can implement this feature?

**Comment by QuincyLarson** [comment]
After discussion on Skype with @SaintPeter @hallaathrad @bugron we are going to remove this completely and implement the "copy as pretty code" button as a standalone button in a forthcoming PR.

**Comment by bugron** [comment]
Reopening. I'll remove this feature in this PR.

**Comment by bugron** [comment]
@QuincyLarson updated. I left Clipboard.js in bower.json as we'll use it later to implement the copy button.

**Comment by M1kep** [comment]
Kinda confused on this stuff, but when will this be implemented?

**Comment by hallaathrad** [comment]
As soon as it gets tested and approved to merge,@M1kep.
Probably today or so. It's been a crazy week ;)

File: client/commonFramework/create-editor.js [hunk 2]
```diff
@@ -37,120 +36,6 @@ window.common = (function(global) {
     }
   );
 
-  var clipboard = new Clipboard('.copy-btn', {
-    text: function(trigger) {
```
Comment (BerkeleyTrue) [inline client/commonFramework/create-editor.js]: In the future, these additions should be done in their own modules.

Comment (bugron) [inline client/commonFramework/create-editor.js]: Do you mean something like creating a new `editor-clipboard.js` file in `client/commonFramework/` and moving `var clipboard = new Clipboard(...` code block there?

**Pull request merged**